gprofiler/utils/__init__.py

```diff
@@ -125,20 +134,25 @@ def start_process(
             env = env if env is not None else os.environ.copy()
             env.update({"LD_LIBRARY_PATH": ""})
 
-    cur_preexec_fn = kwargs.pop("preexec_fn", os.setpgrp)
+    # Disabled because os.setpgrp is not supported on Windows
+    #cur_preexec_fn = kwargs.pop("preexec_fn", os.setpgrp)
+    cur_preexec_fn = kwargs.pop("preexec_fn", None)
+    # End edit
 
     if term_on_parent_death:
         cur_preexec_fn = wrap_callbacks([set_child_termination_on_parent_death, cur_preexec_fn])
 
     popen = Popen(
-        cmd,
-        stdout=kwargs.pop("stdout", subprocess.PIPE),
-        stderr=kwargs.pop("stderr", subprocess.PIPE),
-        stdin=subprocess.PIPE,
-        preexec_fn=cur_preexec_fn,
-        env=env,
-        **kwargs,
-    )
+                cmd,
+                stdout=kwargs.pop("stdout", subprocess.PIPE),
+                stderr=kwargs.pop("stderr", subprocess.PIPE),
+                stdin=subprocess.PIPE,
+                # Disabled for lack of Windows support
+                #preexec_fn=cur_preexec_fn,
```

Comment:
Windows doesn't support `preexec_fn`? Or it's our preexec function that's not Windows-compatible? We can avoid calling it in that case, or refrain from passing it only if it's Windows (and `assert` that no preexec function was passed)

Reply:
Your preexec function (os.setpgrp) is not Windows compatible. I have the is_windows() check to call it without one if it is running on a Windows host system.